src/commands/signzone.rs

```diff
@@ -1248,12 +1249,23 @@ impl SignZone {
             // we skip that, and we skip RRSIGs as they are output only after
             // the RRset that they cover.
             if self.order_rrsigs_after_the_rtype_they_cover {
-                for rrset in owner_rrs
-                    .rrsets()
-                    .filter(|rrset| !matches!(rrset.rtype(), Rtype::SOA | Rtype::RRSIG))
-                {
+                for rrset in owner_rrs.rrsets().filter(|rrset| {
+                    !(matches!(rrset.rtype(), Rtype::SOA | Rtype::RRSIG)
+                    // If run as ldns-signzone we want to list the NSEC RR
+                    // at the end of the RRset of the apex. By default,
+                    // the NSEC RR would preceed the DNSKEY RRset, so we
+                    // need to filter it out here to manually reinsert it
+                    // later. This is only necessary for the NSEC RR at
+                    // the apex, as the ordering issue doesn't appear at
+                    // other locations than the apex.
+                        || (self.invoked_as_ldns
+                            && !self.use_nsec3
```

Comment:
`!self.use_nsec3` is redundant here as if it's using NSEC3 it won't match Rtype::NSEC, so you can remove this line.